test/fixtures/simple-app/boot/promise.reject

```diff
@@ -0,0 +1,8 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+module.exports = function(app) {
+  return Promise.reject('Promise.reject()')
```

Comment:
I am using a bit of a workaround with `scriptExtensions` set to `'.reject'` for this test, so the failing boot script does not get loaded in all other tests, but I can then explicitly request it for that single test and produce the rejection. It's not super clean but I couldn't think of another way to achieve this goal without creating another full project... I hope it's fine?